apps/site/lib/site_web/controllers/schedule/line_api.ex

```diff
@@ -87,7 +82,25 @@ defmodule SiteWeb.ScheduleController.LineApi do
       end)
       |> Enum.into(%{})
 
-    json(conn, combined_data_by_stop)
+    Jason.encode!(combined_data_by_stop)
+  end
+
+  @doc """
+  Provides predictions and vehicle information for a given route and direction, organized by stop.
+  The line diagram polls this endpoint for its real-time data.
+  """
+  @spec realtime(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def realtime(conn, %{"id" => route_id, "direction_id" => direction_id}) do
+    realtime_key = [route_id, direction_id, conn.assigns.date]
```

Comment:
`cache_key` I think would be a more intention-revealing name here; and I think we should make it a tuple (since those are generally preferred for fixed-length collections that are not intended to be iterated)